internal/connector/internal/services/connector_cluster.go

```diff
@@ -396,11 +396,20 @@ func (k *connectorClusterService) GetClientID(clusterID string) (string, error)
 // SaveDeployment creates a connector deployment in the database
 func (k *connectorClusterService) SaveDeployment(ctx context.Context, resource *dbapi.ConnectorDeployment) *errors.ServiceError {
 	dbConn := k.connectionFactory.New()
+	if resource.Version != 0 {
+		dbConn = dbConn.Where("version = ?", resource.Version)
+	}
 
-	if err := dbConn.Save(resource).Error; err != nil {
+	saveResult := dbConn.Save(resource)
```

Comment:
I worry about this.

See https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1637/files#diff-aee57c287964506015af5e0e2839d1e78a19635c9ce232369e8be08a91101d89R508-R517

The behaviour of `Save` has changed in newer versions of `gorm`.

I believe PR #1717 and #1637 are effectively in a "race condition".

We should agree which should merge first and then check the other is unaffected.

Reply:
As stated in the more general comment I would vote for merging this one since it fixes tests and current behavior and then reason about how to deal with new gorm version behavior.